File: BotTemplate/src/main/java/com/freya02/bot/commands/text/TextExit.java
package com.freya02.bot.commands.text;

import com.freya02.botcommands.api.Logging;
import com.freya02.botcommands.api.annotations.Optional;
import com.freya02.botcommands.api.annotations.RequireOwner;
import com.freya02.botcommands.api.commands.prefixed.BaseCommandEvent;
import com.freya02.botcommands.api.commands.prefixed.TextCommand;
import com.freya02.botcommands.api.commands.prefixed.annotations.Hidden;
import com.freya02.botcommands.api.commands.prefixed.annotations.JDATextCommand;
import com.freya02.botcommands.api.commands.prefixed.annotations.TextOption;
import org.slf4j.Logger;

public class TextExit extends TextCommand {
	private static final Logger LOGGER = Logging.getLogger();

	@Hidden
	@RequireOwner
	@JDATextCommand(name = "exit")
	public void exit(BaseCommandEvent event, @TextOption @Optional String reason) {
		LOGGER.warn("Exiting for reason: {}", reason);

		event.reactSuccess()
				.mapToResult()
				.queue(x -> System.exit(0));
	}
}

File: examples/src/main/kotlin/io/github/freya022/bot/commands/text/TextExit.kt
package io.github.freya022.bot.commands.text

import io.github.freya022.botcommands.api.commands.annotations.Command
import io.github.freya022.botcommands.api.commands.annotations.RequireOwner
import io.github.freya022.botcommands.api.commands.text.BaseCommandEvent
import io.github.freya022.botcommands.api.commands.text.TextCommand
import io.github.freya022.botcommands.api.commands.text.annotations.Hidden
import io.github.freya022.botcommands.api.commands.text.annotations.JDATextCommand
import io.github.freya022.botcommands.api.commands.text.annotations.TextOption
import io.github.oshai.kotlinlogging.KotlinLogging
import kotlin.system.exitProcess

private val logger = KotlinLogging.logger { }

@Command
class TextExit : TextCommand() {
    @Hidden
    @RequireOwner
    @JDATextCommand(path = ["exit"])
    fun onTextExit(event: BaseCommandEvent, @TextOption reason: String?) {
        logger.warn { "Exiting for reason: $reason" }

        event.reactSuccess()
            .mapToResult()
            .queue { exitProcess(0) }
    }
}